src/west/_bootstrap/main.py

```diff
@@ -231,16 +221,36 @@ def bootstrap(args):
     # Clone the west source code and the manifest into west/. Git will create
     # the west/ directory if it does not exist.
 
-    clone('west repository', west_url, west_rev,
-          os.path.join(directory, WEST_DIR, WEST))
-
     clone('manifest repository', manifest_url, manifest_rev,
           os.path.join(directory, WEST_DIR, MANIFEST))
 
+    # Parse the manifest and look for a section named "west"
+    manifest_file = os.path.join(directory, WEST_DIR, MANIFEST, 'default.yml')
+    with open(manifest_file, 'r') as f:
+        data = yaml.safe_load(f.read())
+
+    if 'west' in data:
+        wdata = data['west']
+        try:
+            pykwalify.core.Core(
+                source_data=wdata,
+                schema_files=[_SCHEMA_PATH]
+            ).validate()
+        except pykwalify.errors.SchemaError as e:
+            sys.exit("Error: Failed to parse manifest file '{}': {}"
+                      .format(manifest_file, e))
+
+        west_url = wdata['url']
```

Comment:
My opinion on this is that it is clearer to users of `west` if those are required fields, and it allows us to loosen in future, ie. making them optional, without breaking anything.

Making them optional now, and later deciding to make them required, is much harder.
But besides that, i'm fine with making the optional, if we believe that is ok.

Reply:
I align with @mbolivar here after having thought about it. I already made this optional in `manifest.py` and I do think we should allow only setting one of the two options. Since the only way to do that is to make them both optional, my preference is to do so.